android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java

```diff
@@ -106,14 +106,7 @@ public boolean getLocationServicesEnabled()
 			}
 		}
 
-		// don't count the passive provider
-		for(String name : providerNames) {
-			if (name.equals(LocationManager.NETWORK_PROVIDER) || name.equals(LocationManager.GPS_PROVIDER)) {
-				return true;
-			}
-		}
-
-		return false;
+		return providerNames.size()!=0 & getLastKnownLocation()!=null;
```

Comment:
```Java
return providerNames.size() > 0 && getLastKnownLocation() != null;
```

Space the conditional operators and use a logical `&&` instead of a bitwise `&` (even though the bitwise & is still valid in this case)